R/env.R

```diff
@@ -574,29 +676,30 @@ env_get <- function(env = caller_env(), nm, inherit = FALSE) {
 #' vector.
 #' All names are returned, even those starting with a dot.
 #'
-#' @inheritParams env
+#' @inheritParams get_env
 #' @return A character vector of object names.
 #' @export
 #' @examples
-#' env_names()
+#' env <- env(a = 1, b = 2)
+#' env_names(env)
 #' @useDynLib rlang rlang_unescape_character
-env_names <- function(env = caller_env()) {
-  env_ <- get_env(env)
-  x <- names(env_)
-  .Call(rlang_unescape_character, x)
+env_names <- function(env) {
```

Comment:
```r
#' @section Names of symbols and objects:
#'
#' Technically, objects are bound to symbols rather than strings,
#' since the R interpreter evaluates symbols (see [is_expr()] for a
#' discussion of symbolic objects versus literal objects). However it
#' is often more convenient to work with strings. In rlang
#' terminology, the string corresponding to a symbol is called the
#' _name_ of the symbol (or by extension the name of an object bound
#' to a symbol).
#'
#' @section Encoding:
#'
#' Note that there are deep encoding issues when you convert a string
#' to symbol and vice versa. Symbols are _always_ in the native
#' encoding (see [set_chr_encoding()]). If that encoding (let's say
#' latin1) cannot support some characters, these characters are
#' serialised to ASCII. That's why you sometimes see strings looking
#' like `<U+1234>`, especially if you're running Windows (as R doesn't
#' support UTF-8 as native encoding on that platform).
#'
#' To alleviate some of the encoding pain, `env_names()` always
#' returns a UTF-8 character vector (which is fine even on Windows)
#' with unicode points unserialised.
```

cc @krlmlr